mmdeploy/codebase/mmaction/deploy/video_recognition.py

```diff
@@ -0,0 +1,201 @@
+            if trans['type'] == 'Resize':
+                has_resize = True
+                keep_ratio = trans.get('keep_ratio', True)
+                scale = trans.scale
+            if trans['type'] in ['TenCrop', 'CenterCrop', 'ThreeCrop']:
+                has_crop = True
+                crop_size = trans.crop_size
+
+        if has_crop and tuple(input_shape) != (crop_size, crop_size):
+            logger.error(
+                f'`input shape` should be equal to `crop_size`: {crop_size},'
+                f' but given: {input_shape}')
+        if has_resize and (not has_crop):
+            if keep_ratio:
+                logger.error('Resize should set `keep_ratio` to False'
+                             ' when `input shape` is given.')
+            if tuple(input_shape) != scale:
+                logger.error(
+                    f'`input shape` should be equal to `scale`: {scale},'
+                    f' but given: {input_shape}')
+    return cfg
+
+
+@MMACTION_TASK.register_module(Task.VIDEO_RECOGNITION.value)
+class VideoRecognition(BaseTask):
+    """VideoRecognition task class.
+
+    Args:
+        model_cfg (Config): Original PyTorch model config file.
+        deploy_cfg (Config): Deployment config file or loaded Config
+            object.
+        device (str): A string represents device type.
+    """
+
+    def __init__(self, model_cfg: mmengine.Config, deploy_cfg: mmengine.Config,
+                 device: str):
+        super(VideoRecognition, self).__init__(model_cfg, deploy_cfg, device)
+
+    def build_backend_model(self,
+                            model_files: Sequence[str] = None,
+                            **kwargs) -> torch.nn.Module:
+        """Initialize backend model.
+
+        Args:
+            model_files (Sequence[str]): Input model files.
+
+        Returns:
+            nn.Module: An initialized backend model.
+        """
+        from .video_recognition_model import build_video_recognition_model
+        model = build_video_recognition_model(
+            model_files, self.model_cfg, self.deploy_cfg, device=self.device)
+        model.to(self.device)
+        model.eval()
+        return model
+
+    def create_input(self,
+                     imgs: Union[str, np.ndarray],
+                     input_shape: Sequence[int] = None,
+                     data_preprocessor: Optional[BaseDataPreprocessor] = None)\
+            -> Tuple[Dict, torch.Tensor]:
+        """Create input for video recognition.
+
+        Args:
+            imgs (str | np.ndarray): Input image(s), accepted data type are
+                `str`, `np.ndarray`.
+            input_shape (list[int]): A list of two integer in (width, height)
+                format specifying input shape. Defaults to `None`.
+
+        Returns:
+            tuple: (data, img), meta information for the input image and input.
+        """
+        if isinstance(imgs, (list, tuple)):
+            if not isinstance(imgs[0], str):
```

Comment:
Use `all(isinstance(img, str) for img in imgs)` to check all imgs instead of the first one.